lms/djangoapps/course_blocks/transformers/user_partitions.py

```diff
@@ -201,7 +172,7 @@ def transform(self, user_info, block_structure):
         if not user_partitions:
             return
 
-        user_groups = self._get_user_partition_groups(
+        user_groups = get_user_partition_groups(
             user_info.course_key, user_partitions, user_info.user
         )
         if not user_info.has_staff_access:
```

Comment:
For efficiency, move the check for *not* staff_access to the line above `get_user_partition_groups`.